contracts/TokenIOStorage.sol

```diff
@@ -647,9 +656,87 @@ contract TokenIOStorage is Ownable {
      * @param _address Pointer identifier for value in mapping
      * @return { "uint" : "Returns the uint value associated with the key" }
      */
-    function deleteTokenfxUSDBPSRate(address _address) external view returns(bool success) {
+    function deleteTokenfxUSDBPSRate(address _address) external onlyOwner returns(bool success) {
         delete assets[_address].fxUSDBPSRate;
         return true;
     }
 
+
+    function setFirm(string firmName, bool approved) external onlyOwner returns (bool success) {
+      firms.push(Firm(firmName, address(0), approved));
+      return true;
+    }
+
+    function getFirmDetails(string firmName) external view returns(address authorityAddress, bool approved) {
+        for(uint i = 0; i < firms.length; i++) {
```

Comment:
undefined amount of gas used. Potential point of failure. Let's avoid this array entirely

Reply:
Yes it's a concept problem. Need to discuss